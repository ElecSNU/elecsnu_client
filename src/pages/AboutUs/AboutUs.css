#about-us {
    height: 100%;
    width: 100%;
    display: grid;
    place-items: center;
}

.about-us-container {
    width: 75%;
    height: 80%;
    padding: 2.5%;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
}

.about-us-heading {
    padding: 1% 2.5%;
    width: 100%;
    font-size: 2.5rem;
}

.teammates {
    width: 100%;
    height: 70%;
    margin-top: 2.5%;
    display: flex;
    justify-content: space-between;
}

.teammate-card {
    width: 22.5%;
    height: 100%;
    margin: 0 1.25%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-evenly;
}

.divider {
    height: 2.5px;
    width: 75%;
    background-color: gray;
}
