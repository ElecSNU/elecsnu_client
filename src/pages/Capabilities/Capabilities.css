#capabilities-page {
    height: 100%;
    width: 100%;
    display: grid;
    place-items: center;
    overflow-y: scroll;
}

.capabilities-container {
    height: 80vh;
    margin: 6.825vh 0;
    width: 85%;
    padding: 2% 2.5%;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
}

#capabilities-page .main-heading {
    font-size: 3.5rem;
}

.capabilities-links {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.capability-link {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.capability-link .img-container {
    padding: 2.25rem;
    margin-bottom: 1.5rem;
    background-color: #004aad;
    border-radius: 100rem;
}

.capability-link img {
    margin: 0;
    width: 7.5rem;
}

.capability {
    height: 90%;
    display: flex;
    margin: 0;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-between;
    padding: 2.5%;
}

.capability-headings {
    flex: 1;
    width: 22.5%;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-evenly;
}

.capability-points {
    flex: 3;
    width: 72.5%;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    justify-content: space-between;
    list-style: none;
}

.capability-points li {
    margin-bottom: 1%;
    text-align: right;
}

.capability-points p {
    text-align: right;
}
