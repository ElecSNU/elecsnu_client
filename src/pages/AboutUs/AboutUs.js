import React from 'react';

import Megha from '../../assets/images/megha.png';
import Ishit from '../../assets/images/ishit.png';
import Yashi from '../../assets/images/yashi.png';
import Kaustubh from '../../assets/images/kaustubh.png';

import './AboutUs.css';

const TeamMateCard = ({ details }) => {
    return (
        <div
            className={`teammate-card background-${details.background} border-radius-15 box-shadow`}
        >
            <img
                src={details.image}
                alt=''
                className='teammate-img'
            />
            <h1 className='teammate-name heading-text capital-text foreground-light'>
                {details.name}
            </h1>
            <div className='divider'></div>
            <h1 className='teammate-name subheading-text capital-text foreground-light'>
                {details.designation}
            </h1>
        </div>
    );
};

const AboutUs = () => {
    const teammates = [
        {
            image: Megha,
            name: 'Megha Agarwal',
            designation: 'DESIGNER',
            background: 'accent1',
        },
        {
            image: Ishit,
            name: 'Ishit Beswal',
            designation: 'Coder',
            background: '1',
        },
        {
            image: Yashi,
            name: 'Yashi Jain',
            designation: 'Coder',
            background: '1',
        },
        {
            image: Kaustubh,
            name: 'Kaustubh Rai',
            designation: 'Tester',
            background: 'accent2    ',
        },
    ];

    return (
        <div id='about-us'>
            <div className='about-us-container border-radius-15 box-shadow background-light-translucent'>
                <div className='about-us-heading heading-text background-dark border-radius-15 foreground-light center-text'>
                    OUR TEAM
                </div>
                <div className='teammates'>
                    {teammates.map((teammate, index) => {
                        return (
                            <TeamMateCard
                                key={index}
                                details={teammate}
                            />
                        );
                    })}
                </div>
            </div>
        </div>
    );
};

export default AboutUs;
