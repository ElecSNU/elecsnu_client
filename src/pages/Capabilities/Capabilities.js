import React from 'react';
import { Link } from 'react-router-dom';

import './Capabilities.css';

const Capabilities = () => {
    const capabilities = [
        {
            id: 'integrity',
            heading: 'Voting Integrity',
            subheading:
                'Trust the results of your votes and elections',
            points: [
                {
                    heading: 'One-to-One Voting',
                    text:
                        'Ensure voters are not able to cast multiple votes on the same ballot',
                },
                {
                    heading: 'Voter Authentication',
                    text:
                        'The voters are authenticated using Machine learning techniques like face detection and face recognition',
                },
            ],
        },
        {
            id: 'security',
            heading: 'Security',
            subheading:
                "Ensure the election's security and peace of mind.",
            points: [
                {
                    heading: 'Anonymity of votes',
                    text:
                        'Even the admin will not be able to see who voted for whom. Thus maintaining the anonymity of votes.',
                },
                {
                    heading: 'Voter Login',
                    text:
                        'Face detection technique is used to make sure only one person logs in while voting. Face recognition is also carried out to ensure that the person casting the vote and the one logged in are the same ',
                },
            ],
        },
        {
            id: 'admin',
            heading: 'Admin Experience',
            subheading:
                'Manage the voting process with ease',
            points: [
                {
                    heading: 'Voting Platform',
                    text:
                        'Gain control over your voting events. See the results analysis and everything at one place',
                },
                {
                    heading: 'Vote Management Services',
                    text:
                        'The admin can create groups of voters for different elections like club elections, student council elections, etc.',
                },
                {
                    heading: 'Voter Upload Modules',
                    text:
                        'Our voter list upload modules let you manage the list of eligible voters, their details, and login information.',
                },
                {
                    heading: 'Vote Scheduling Options',
                    text:
                        'Set elections to open and close at a preferred date and time',
                },
            ],
        },
        {
            id: 'voter',
            heading: 'Voter Experience',
            subheading:
                'Offer voters a straightforward, pleasant experience',
            points: [
                {
                    heading: 'Voter Notifications',
                    text:
                        'Even the admin will not be able to see who voted for whom. Thus maintaining the anonymity of votes.',
                },
                {
                    heading:
                        'Effective and User Friendly UI',
                    text:
                        'Providing voters an alternate login option using their college ID cards which is a unique-to-the-voter link that bypasses the traditional login page.',
                },
                {
                    heading: 'Candidate Manifestos',
                    text:
                        'There is an option to view the manifestos of all the candidates so that voters can make a more informed decision',
                },
                {
                    heading: 'Social Media',
                    text:
                        'In order to increase the voter turnout,we provide an option for the people who voted to upload a picture on their social media',
                },
            ],
        },
    ];

    return (
        <div id='capabilities'>
            <div className='capabilities-container background-light-translucent border-radius-15 box-shadow'>
                <h1 className='heading-text main-heading foreground-back capital-text center-text'>
                    Capabilities
                </h1>
                <div className='capabilities-links'>
                    {capabilities.map(
                        (capability, index) => (
                            <a
                                className='capability-link'
                                key={index}
                                href={`#${capability.id}`}
                            >
                                <img src='' alt='' />
                                <h1 className='heading-text foreground-accent1 capital-text center-text'>
                                    {capability.heading}
                                </h1>
                            </a>
                        )
                    )}
                </div>
            </div>
            {capabilities.map((capability, index) => (
                <>
                    <div
                        id={capability.id}
                        className={`capabilities-container border-radius-15 box-shadow capability foreground-${
                            index % 2 !== 0
                                ? 'back'
                                : 'light'
                        } background-${
                            index % 2 !== 0
                                ? 'light-translucent'
                                : '1-translucent'
                        }`}
                        key={index}
                    >
                        <div className='capability-headings'>
                            <h1 className='heading-text capital-text'>
                                {capability.heading}
                            </h1>
                            <h2 className='subheading-text'>
                                {capability.subheading}
                            </h2>
                        </div>
                        <ul className='capability-points'>
                            {capability.points.map(
                                (capabilityPoint, i) => (
                                    <li key={i}>
                                        <div className='subheading-text'>
                                            {
                                                capabilityPoint.heading
                                            }
                                        </div>
                                        <p className='normal-text'>
                                            {
                                                capabilityPoint.text
                                            }
                                        </p>
                                    </li>
                                )
                            )}
                        </ul>
                    </div>
                    <br />
                    <br />
                </>
            ))}
            <div style={{ opacity: '0' }}>.</div>
        </div>
    );
};

export default Capabilities;
