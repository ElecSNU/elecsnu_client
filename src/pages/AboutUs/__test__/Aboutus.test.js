import { render, screen, fireEvent, cleanup } from '@testing-library/react';
    import ReactDOM from 'react-dom'
    // import React , { useState }from 'react'
    import renderer from 'react-test-renderer'
import AboutUs from './../AboutUs';
    // import DesignSide from './../DesignSide'
    import Capabilities from './../AboutUs'

    test('Tests the rendering of AboutUs', () => {
        const div = document.createElement("div")
        const tree = renderer.create(<AboutUs></AboutUs>).toJSON();
        expect(tree).toMatchSnapshot();
       });