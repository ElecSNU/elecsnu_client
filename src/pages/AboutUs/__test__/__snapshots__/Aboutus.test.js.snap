// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the rendering of AboutUs 1`] = `
<div
  id="about-us"
>
  <div
    className="about-us-container border-radius-15 box-shadow background-light-translucent"
  >
    <div
      className="about-us-heading heading-text background-dark border-radius-15 foreground-light center-text"
    >
      OUR TEAM
    </div>
    <div
      className="teammates"
    >
      <div
        className="teammate-card background-accent1 border-radius-15 box-shadow"
      >
        <img
          alt=""
          className="teammate-img"
          src="megha.png"
        />
        <h1
          className="teammate-name heading-text capital-text foreground-light"
        >
          Megha Agarwal
        </h1>
        <div
          className="divider"
        />
        <h1
          className="teammate-name subheading-text capital-text foreground-light"
        >
          DESIGNER
        </h1>
      </div>
      <div
        className="teammate-card background-1 border-radius-15 box-shadow"
      >
        <img
          alt=""
          className="teammate-img"
          src="ishit.png"
        />
        <h1
          className="teammate-name heading-text capital-text foreground-light"
        >
          Ishit Beswal
        </h1>
        <div
          className="divider"
        />
        <h1
          className="teammate-name subheading-text capital-text foreground-light"
        >
          Coder
        </h1>
      </div>
      <div
        className="teammate-card background-1 border-radius-15 box-shadow"
      >
        <img
          alt=""
          className="teammate-img"
          src="yashi.png"
        />
        <h1
          className="teammate-name heading-text capital-text foreground-light"
        >
          Yashi Jain
        </h1>
        <div
          className="divider"
        />
        <h1
          className="teammate-name subheading-text capital-text foreground-light"
        >
          Coder
        </h1>
      </div>
      <div
        className="teammate-card background-accent2     border-radius-15 box-shadow"
      >
        <img
          alt=""
          className="teammate-img"
          src="kaustubh.png"
        />
        <h1
          className="teammate-name heading-text capital-text foreground-light"
        >
          Kaustubh Rai
        </h1>
        <div
          className="divider"
        />
        <h1
          className="teammate-name subheading-text capital-text foreground-light"
        >
          Tester
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests the rendering of Capabilities 1`] = `
<div
  id="about-us"
>
  <div
    className="about-us-container border-radius-15 box-shadow background-light-translucent"
  >
    <div
      className="about-us-heading heading-text background-dark border-radius-15 foreground-light center-text"
    >
      OUR TEAM
    </div>
    <div
      className="teammates"
    >
      <div
        className="teammate-card background-accent1 border-radius-15 box-shadow"
      >
        <img
          alt=""
          className="teammate-img"
          src="megha.png"
        />
        <h1
          className="teammate-name heading-text capital-text foreground-light"
        >
          Megha Agarwal
        </h1>
        <div
          className="divider"
        />
        <h1
          className="teammate-name subheading-text capital-text foreground-light"
        >
          DESIGNER
        </h1>
      </div>
      <div
        className="teammate-card background-1 border-radius-15 box-shadow"
      >
        <img
          alt=""
          className="teammate-img"
          src="ishit.png"
        />
        <h1
          className="teammate-name heading-text capital-text foreground-light"
        >
          Ishit Beswal
        </h1>
        <div
          className="divider"
        />
        <h1
          className="teammate-name subheading-text capital-text foreground-light"
        >
          Coder
        </h1>
      </div>
      <div
        className="teammate-card background-1 border-radius-15 box-shadow"
      >
        <img
          alt=""
          className="teammate-img"
          src="yashi.png"
        />
        <h1
          className="teammate-name heading-text capital-text foreground-light"
        >
          Yashi Jain
        </h1>
        <div
          className="divider"
        />
        <h1
          className="teammate-name subheading-text capital-text foreground-light"
        >
          Coder
        </h1>
      </div>
      <div
        className="teammate-card background-accent2     border-radius-15 box-shadow"
      >
        <img
          alt=""
          className="teammate-img"
          src="kaustubh.png"
        />
        <h1
          className="teammate-name heading-text capital-text foreground-light"
        >
          Kaustubh Rai
        </h1>
        <div
          className="divider"
        />
        <h1
          className="teammate-name subheading-text capital-text foreground-light"
        >
          Tester
        </h1>
      </div>
    </div>
  </div>
</div>
`;
